lib/hdf5-db/jhdf5-db.c

```diff
@@ -0,0 +1,381 @@
+
+#include <hdf5.h>
+#include <H5PLextern.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+
+static
+herr_t
+H5VL_julea_db_attr_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_attr_term(void);
+static
+herr_t
+H5VL_julea_db_dataset_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_dataset_term(void);
+static
+herr_t
+H5VL_julea_db_datatype_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_datatype_term(void);
+static
+herr_t
+H5VL_julea_db_file_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_file_term(void);
+static
+herr_t
+H5VL_julea_db_group_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_group_term(void);
+static
+herr_t
+H5VL_julea_db_space_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_space_term(void);
+static
+herr_t
+H5VL_julea_db_link_init(hid_t vipl_id);
+static
+herr_t
+H5VL_julea_db_link_term(void);
+
+#include "jhdf5-db.h"
+#include "jhdf5-db-shared.c"
+#include "jhdf5-db-file.c"
+#include "jhdf5-db-dataset.c"
+#include "jhdf5-db-attr.c"
+#include "jhdf5-db-group.c"
+#include "jhdf5-db-datatype.c"
+#include "jhdf5-db-space.c"
+#include "jhdf5-db-link.c"
+
+#define _GNU_SOURCE
+
+#define JULEA_DB 530
+
+static
+herr_t
+H5VL_julea_db_init(hid_t vipl_id)
+{
+	J_TRACE_FUNCTION(NULL);
+
+	if (H5VL_julea_db_file_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_file;
+	}
+	if (H5VL_julea_db_dataset_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_dataset;
+	}
+	if (H5VL_julea_db_attr_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_attr;
+	}
+	if (H5VL_julea_db_datatype_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_datatype;
+	}
+	if (H5VL_julea_db_group_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_group;
+	}
+	if (H5VL_julea_db_space_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_space;
+	}
+	if (H5VL_julea_db_link_init(vipl_id))
+	{
+		G_DEBUG_HERE();
+		goto _error_link;
+	}
+
+	return 0;
+_error_link:
+	H5VL_julea_db_link_term();
+_error_space:
+	H5VL_julea_db_space_term();
+_error_group:
+	H5VL_julea_db_group_term();
+_error_datatype:
+	H5VL_julea_db_datatype_term();
+_error_attr:
+	H5VL_julea_db_attr_term();
+_error_dataset:
+	H5VL_julea_db_dataset_term();
+_error_file:
+	H5VL_julea_db_file_term();
+
+	return 1;
+}
+
+static
+herr_t
+H5VL_julea_db_term(void)
+{
+	J_TRACE_FUNCTION(NULL);
+
+	if (H5VL_julea_db_link_term())
+	{
+		j_goto_error();
+	}
+	if (H5VL_julea_db_space_term())
+	{
+		j_goto_error();
+	}
+	if (H5VL_julea_db_group_term())
+	{
+		j_goto_error();
+	}
+	if (H5VL_julea_db_datatype_term())
+	{
+		j_goto_error();
+	}
+	if (H5VL_julea_db_attr_term())
+	{
+		j_goto_error();
+	}
+	if (H5VL_julea_db_dataset_term())
+	{
+		j_goto_error();
+	}
+	if (H5VL_julea_db_file_term())
+	{
+		j_goto_error();
+	}
+	return 0;
+_error:
+	return 1;
+}
+
+/**
+ * The class providing the functions to HDF5
+ * @see dependencies/opt/spack/linux-ubuntu19.04-x86_64/gcc-8.3.0/hdf5-develop-4iami4kalqj7xgv2x2uv25dnzvz4xzwf/include/H5VLconnector.h
```

Comment:
This path will be different for each user.

Reply:
Yes, but the path-structure will be the same.
This is just meant as a hint for the reader, where the hdf-vol-plugin structure is defined.

Reply:
Copying the file is dangerous, because the dependency may change in the future